### DISCUSSION TITLE
ONLINE PYGAME EXECUTION

### DISCUSSION THREAD

**Discussion by cool-dev-guy** [pygame-web/pygbag#99] [Q&A] [answered]
Hey @pmp-p ,I studied the pygame-web final product of a pygame project and i understood that first there is a GET request to `game_name.apk` and inside it in the `assets` folder i can find a `main.py`,Is it possible to change the code inside `main.py` before its being executed,Or can i make the website run the python code which i stored in a javascript/python variable?

**Comment by pmp-p** [comment, marked as answer]
To run arbitrary python code from javascript use Module.PyRun_SimpleString(code:str). ( coming from https://docs.python.org/3/c-api/veryhigh.html )

To run arbitrary python code from html use pygame-script, various samples here https://github.com/pygame-web/showroom/tree/main/pygame-scripts
